obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+        new_event.origins[0].arrivals.append(Arrival(phase=new_event.
+                                                     picks[pick_index].
+                                                     phase_hint,
+                                                     pick_id=new_event.
+                                                     picks[pick_index].
+                                                     resource_id))
+        if weight != 999:
+            new_event.origins[0].arrivals[pick_index].time_weight =\
+                weight
+        if azimuthres != 999:
+            new_event.origins[0].arrivals[pick_index].backazimuth_residual =\
+                azimuthres
+        if timeres != 999:
+            new_event.origins[0].arrivals[pick_index].time_residual =\
+                timeres
+        if distance != 999:
+            new_event.origins[0].arrivals[pick_index].distance =\
+                distance
+        if caz != 999:
+            new_event.origins[0].arrivals[pick_index].azimuth =\
+                caz
+    # Write event to catalog object for ease of .write() method
+    return new_event
+
+
+def readwavename(sfile):
+    """
+    Extract the waveform filename from the s-file.
+    Returns a list of waveform names found in the s-file as multiples can \
+    be present.
+
+    :type sfile: str
+    :param sfile: Path to the sfile
+
+    :returns: List of strings of wave paths
+    :rtype: list
+    """
+    f = open(sfile, 'r')
+    wavenames = _readwavename(f=f)
+    f.close()
+    return wavenames
+
+
+def _readwavename(f):
+    """
+    Internal wavename reader.
+
+    :type f: file
+    :param f: File open in read-mode
+    :return: list of wavefile names
+    """
+    wavename = []
+    for line in f:
+        if len(line) == 81 and line[79] == '6':
+            wavename.append(line[1:79].strip())
+    return wavename
+
+
+def blanksfile(wavefile, evtype, userid, outdir, overwrite=False,
+               evtime=False):
+    """
+    Generate an empty s-file with a populated header for a given waveform.
+
+    :type wavefile: str
+    :param wavefile: Wavefile to associate with this S-file, the timing of \
+        the S-file will be taken from this file if evtime is not set.
+    :type evtype: str
+    :param evtype: Event type letter code, e.g. L, R, D
+    :type userid: str
+    :param userid: 4-character SEISAN USER ID
+    :type outdir: str
+    :param outdir: Location to write S-file
+    :type overwrite: bool
+    :param overwrite: Overwrite an existing S-file, default=False
+    :type evtime: obspy.core.utcdatetime.UTCDateTime
+    :param evtime: If given this will set the timing of the S-file
+
+    :returns: str, S-file name
+    """
+
+    from obspy import read as obsread
+    import os
+    import datetime
+
+    if not evtime:
+        try:
+            st = obsread(wavefile)
+            evtime = st[0].stats.starttime
+        except:
+            raise IOError('Wavefile: ' + wavefile +
```

Comment:
Please don't use `IOError` - this error is reserved for Python when an I/O operation like `open()` fails. Use either `ValueError` or just define your own exception, e.g. `NordicSFileParsingError` or something along these lines.